src/redux/books/books.js

```diff
@@ -1,25 +1,42 @@
+import axios from '../../services/axios';
+
+const LIST_BOOKS = 'bookstore/books/LIST_BOOKS';
 const ADD_BOOK = 'bookstore/books/ADD_BOOK';
 const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
 
 export default (state = [], action) => {
   switch (action.type) {
+    case LIST_BOOKS:
+      return action.books;
+
     case ADD_BOOK:
       return [...state, action.book];
 
     case REMOVE_BOOK:
-      return [...state.filter((b) => b.id !== action.bookId)];
+      return [...state.filter((b) => b.item_id !== action.item_id)];
 
     default:
       return state;
   }
 };
 
-export const addBookAction = (book) => ({
-  type: ADD_BOOK,
-  book,
-});
+export const getBooksAction = () => async (dispatch) => {
+  const response = await axios.get('books');
+  const data = Object.keys(response.data).map((book) => ({
+    item_id: book,
+    ...response.data[book][0],
+  }));
+  dispatch({ type: LIST_BOOKS, books: data });
+};
 
-export const removeBookAction = (bookId) => ({
-  type: REMOVE_BOOK,
-  bookId,
-});
+export const addBookAction = (book) => async (dispatch) => {
+  const request = await axios.post('books', book);
+  if (request.status === 201) {
+    dispatch({ type: ADD_BOOK, book });
+  }
+};
+
+export const removeBookAction = (bookId) => async (dispatch) => {
+  const request = await axios.delete(`books/${bookId}`);
+  if (request.status === 201) dispatch({ type: REMOVE_BOOK, item_id: bookId });
+};
```

Comment:
![image](https://user-images.githubusercontent.com/93039359/214791731-c2e831fa-18a2-499d-a7da-f62cd679dc0f.png)

- [ ] Kindly use createAsyncThunk instead of using the traditional ways which are about to be deprecated :thinking: This is a requirement in your project that you are supposed to use createAsyncThunk with fetch or axios so kindly follow this requirement :+1: